Update README.md (#251)
GitGuardian/ggshield: README.md

The package should run on MacOS, Linux and Windows.

#### Updating

To update ggshield you can add the option `-U/--upgrade` to the pip install command.

```shell
$ pip install -U ggshield
```

# Initial setup

To use ggshield you need to authenticate against GitGuardian servers. To do so, use the `ggshield auth login` command. This command automates the provisioning of a personal access token and its configuration on the local workstation.

You can learn more about it from [`ggshield auth login` documentation](https://docs.gitguardian.com/internal-repositories-monitoring/ggshield/reference/auth/login).

> Alternatively, if you already have a personal access token, you can store it in the `GITGUARDIAN_API_KEY` environment variable.

Reviewer: ```suggestion
Alternatively, you can create your personal access token manually and you can store it in the `GITGUARDIAN_API_KEY` environment variable to complete the setup.
```

## Environment Variables

ggshield can also be configured using environment variables.

Environment variables overrides settings set on your configuration file but are overridden by command line options.

At startup, ggshield attempts to load environment variables from different environment files in the following order:

- path pointed to by the environment variable `GITGUARDIAN_DOTENV_PATH`
- `.env` at your current work directory.
- `.env` at the root of the current git directory

Only one file will be loaded of the three.

Reference of current environment variables supported by ggshield:

- `GITGUARDIAN_API_KEY`: API Key for the GitGuardian API. Use this if you don't want to use the `ggshield auth` commands.

> - `GITGUARDIAN_INSTANCE`: Custom URL of the GitGuardian dashboard. The API URL will be inferred from it.

Reviewer: These modifications of env variables need to be modified in public doc as well. Can you open a MR there please?

- `GITGUARDIAN_API_URL`: Custom URL for the scanning API. Deprecated, use `GITGUARDIAN_INSTANCE` instead.

> - `GITGUARDIAN_DONT_LOAD_ENV`: If set to any value then environment variables won't be loaded from a file.

Reviewer: ```suggestion
- `GITGUARDIAN_DONT_LOAD_ENV`: If set to any value, environment variables won't be loaded from a file.
```

- `GITGUARDIAN_DOTENV_PATH`: If set to a path, ggshield will attempt to load the environment from the specified file.

- `GITGUARDIAN_TIMEOUT`: If set to a float, `ggshield secret scan pre-receive` will timeout after the specified value. Set to 0 to disable the timeout.

- `GITGUARDIAN_MAX_COMMITS_FOR_HOOK`: If set to an int, `ggshield secret scan pre-receive` and `ggshield secret scan pre-push` will not scan more than the specified value of commits in a single scan.

- `GITGUARDIAN_CRASH_LOG`: If set to True, ggshield will display a full traceback when crashing.

## On-premises configuration

ggshield can be configured to run on your on-premises dashboard.

> First, you need to point ggshield to your instance, by either defining the `instance` key in your `.gitguardian.yaml` configuration file or by defining the `GITGUARDIAN_INSTANCE` environment variable.

Reviewer: Is it a necessary step? If yes, it needs to be also documented in public doc.

Isn't doing the authentication with `--instance https://myinstance` enough?

Author: Yes, this is required: the value of `auth login --instance` option is used to decide the instance to authenticate against and is stored with the associated PAT, but it does not define the repository instance.

Going to add this to the public doc MR.

> Then, you need to authenticate against your instance, by either using the `ggshield auth` commands or by obtaining an API key from your dashboard administrator and storing it in the `GITGUARDIAN_API_KEY` environment variable.

Reviewer: ```suggestion
Then, you need to authenticate against your instance, by either using the `ggshield auth login --instance https://mygitguardianinstance.mycorp.local` command using the `--instance` option or by obtaining an API key from your dashboard administrator and storing it in the `GITGUARDIAN_API_KEY` environment variable.
```

> You may be interested in using GitGuardian's [GitHub integration](https://dashboard.gitguardian.com/settings/workspace/integrations/github) to ensure full coverage of your GitHub projects as well as full git history scans and reporting.

> ggshield support for GitHub comes in the form of GitHub actions.

Reviewer: this needs to be also corrected in public doc

Author: The public doc for GitHub action is a bit different than this one. I think it looks fine: https://docs.gitguardian.com/internal-repositories-monitoring/integrations/ci_cd_integrations/github_actions.
